node/network/protocol/src/lib.rs

```diff
@@ -336,6 +336,67 @@ pub mod v1 {
 		Statement(Hash, SignedFullStatement)
 	}
 
+	#[derive(Debug, Clone, Copy, PartialEq, thiserror::Error)]
+	#[allow(missing_docs)]
+	pub enum CompressedPoVError {
+		#[error("Failed to compress a PoV")]
+		Compress,
+		#[error("Failed to decompress a PoV")]
+		Decompress,
+		#[error("Failed to decode the uncompressed PoV")]
+		Decode,
+		#[error("Architecture is not supported")]
+		NotSupported,
+	}
+
+	/// SCALE and Zstd encoded [`PoV`].
+	#[derive(Debug, Clone, Encode, Decode, PartialEq)]
+	pub struct CompressedPoV(Vec<u8>);
+
+	impl CompressedPoV {
+		/// Compress the given [`PoV`] and returns a [`CompressedPoV`].
+		#[cfg(not(target_os = "unknown"))]
+		pub fn compress(pov: &PoV) -> Result<Self, CompressedPoVError> {
+			zstd::encode_all(pov.encode().as_slice(), 3).map_err(|_| CompressedPoVError::Compress).map(Self)
```

Comment:
@bkchr , I was wondering if you did test with something like (curious if it gives similar ratio):
```
let mut dest = Vec::new();
pov.encode_to(Encoder::new(&mut dest, 3)
```